core/build.gradle.kts

```diff
@@ -187,6 +192,40 @@ ide {
     generatedSource(javaCCTest, "test")
 }
 
+fun JavaCompile.configureAnnotationSet(sourceSet: SourceSet) {
+    source = sourceSet.java
+    classpath = sourceSet.compileClasspath
+    destinationDirectory.set(options.generatedSourceOutputDirectory)
+    options.compilerArgs.add("-proc:only")
+    org.gradle.api.plugins.internal.JvmPluginsHelper.configureAnnotationProcessorPath(sourceSet, sourceSet.java, options, project)
+}
+
+val annotationProcessorMain by tasks.registering(JavaCompile::class) {
+    dependsOn(javaCCMain)
+    configureAnnotationSet(sourceSets.main.get())
+    // only if we aren't running compileJava, since doing twice fails.
+    onlyIf { tasks.findByPath("compileJava")?.enabled != true }
```

Comment:
`tasks.findByPath("compileJava")?.enabled` is verifying if the task exists, however, what we need to check is whether task is a part of the task graph (e.g. if it was requested by the user).

The condition should be like `project.gradle.taskGraph.hasTask("compileJava")` or something like project.gradle.taskGraph { if (compileJava is in the graph) { disable proc-only task}`.